### ISSUE TITLE
[FEATURE] Enhanced Connectivity and Introduction of JSFunctionChain Node

### ISSUE THREAD

**Issue by gabriellechenco** [FlowiseAI/Flowise#1934]
**Describe the feature you'd like**
I would like to propose an improvement in the connectivity of `Chain` nodes within Flowise, akin to the functionality offered by LangChain. Currently, users face frustration when attempting to connect Chain nodes, which severely limits the available design patterns and constrains the flexibility of solutions beyond the use case templates.

The proposed enhancement involves enabling seamless connectivity between all Chain nodes, facilitating the creation of complex flow patterns. Additionally, the introduction of a new node, `JSFunctionChain`, would serve as an intermediary step between `Chains`, enhancing the extensibility and versatility of workflow designs.

**Additional context**
Given that Flowise utilizes Langchain under the hood, it's frustrating that `Chain` nodes often fail to connect effectively, restricting design options. While the LLMChain offers connectivity, it's limited to specific examples like the "Prompt Chaining" in the Marketplace. Enabling chainability across all Chains would significantly enhance the platform's usability and empower users to create more intricate and dynamic workflows.

Prompt Chaining example:
![image](https://github.com/FlowiseAI/Flowise/assets/87021041/a4f3dcda-9014-48c9-9ec7-63e31094a158)

**Comment by HenryHengZJ** [comment]
Does the CustomFunction node in Flowise fits the purpose?

**Comment by gabriellechenco** [comment]
@HenryHengZJ I don't think so. Cause the Custom Function expect a json as input, so the nodes don't will connect with each other even in this scenario.

When we are use the LangChain framework in the Flowise, is to be expected that the Chains nodes can be connected from left to right, allowing to treat the information and insert the last Chain output in the next Chain Input. Even if was possible with a Custom JS Function, it seems more like a hack than a feature. 

Some nodes interacting with each other and other don't give to Flowise this taste of incompleteness.
